tests/e2e/common/mod.rs

```diff
@@ -144,14 +152,17 @@ impl Config {
 pub(crate) struct TestEnvironment {
     pub(crate) config: &'static Config,
     pub(crate) client: Client,
+    pub(crate) is_local: bool,
 }
 
 pub(crate) fn setup_global() -> TestEnvironment {
     let _ = dotenvy::dotenv();
 
     let _ = env_logger::builder().parse_default_env().is_test(true).try_init();
 
-    TestEnvironment { config: &CONFIG, client: client() }
+    let (is_local, client) = client();
+
+    TestEnvironment { config: &CONFIG, client, is_local }
```

Comment:
Could `is_local` be on `config` rather than directly on `TestEnvironment`? That's what it's meant for